lib/pbench/test/unit/server/test_cache_manager.py

```diff
@@ -339,10 +342,83 @@ def __init__(self, path: Path, cache: Path, logger: Logger):
             self.datasets = {}
             self.tarballs = {}
 
+        @staticmethod
+        def generate_test_result_tree(tmp_path: Path, dir_name: str) -> Path:
+            """
+            Directory Structure
+
+            /tmp/
+                <dir_name>/
+                    subdir1/
+                        subdir11/
+                        subdir12/
+                            f121_sym -> /tmp/<dir_name>/subdir1/subdir15
+                        subdir13/
+                            f131_sym -> /etc/passwd
+                        subdir14/
+                            subdir141/
+                                f1411.txt
+                                f1412_sym -> /tmp/<dir_name>/subdir1/f11.txt
+                        f11.txt
+                    f1.json
```

Comment:
The coverage report for your latest update shows coverage for all but one case (ignoring [CacheManager.get_info()](http://jenkins.intlab.perf-infra.lab.eng.rdu2.redhat.com/job/pbench-ci-multi/job/PR-3396/49/cobertura/server/cache_manager_py/#1004)):  we are still missing the case for [a symlink with a reference which is relative and refers to an existing file inside the results area which is not a file or directory](http://jenkins.intlab.perf-infra.lab.eng.rdu2.redhat.com/job/pbench-ci-multi/job/PR-3396/49/cobertura/server/cache_manager_py/#157).

This case was supposed to be covered by `f1415_sym`, but, unfortunately, we outsmarted ourselves and that symlink doesn't drive the behavior that we want.  (My apologies for the bad suggestion!)  We need `f1415_sym` to be a symlink, containing a relative reference to an existing file, which is not a regular file or directory...so having its target be a symlink would be good, but that symlink has to contain a reference which cannot be resolved.

So, having `f1415_sym` referring to `f1412_sym` doesn't work because, although it points outside the results area, `f1412_sym`'s reference is _resolvable_.  (It basically duplicates the case tested by `f1414_sym`.)

Instead, try having `f1415_sym` refer to `f122_sym` (i.e., to `../../subdir12/f122_sym`).  (Alternately, if you think it's valuable, you can leave `f1415_sym` alone and _add_ another symlink referring to `f122_sym`.)

Reply:
> we are still missing the case for a symlink with a reference which is relative and refers to an existing file inside the results area which is not a file or directory.

Unfortunately, the latest changes didn't close this gap.  The problem is that we are using "strict" path resolution, and so a symlink to a (bad) symlink results in a `FileNotFound` error instead of what I expected.

So, unless we are going to go so far as to create a FIFO or a Unix socket or some other special file, I guess we're going to have to put up with imperfect test coverage.

We could reconsider whether to use strict path resolution (I don't see any need for it, now), but perhaps we should do that in another PR.